.github/workflows/build-deploy-storybook.yml

```diff
@@ -10,22 +10,24 @@ jobs:
     permissions:
       contents: write
     runs-on: ubuntu-latest
+    environment: github-pages
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
         with:
-          # We check out the pull request's base branch, which will be
-          # used as the base branch for all git operations.
-          ref: ${{ github.event.pull_request.base.ref }}
-      - name: Get Node.js version
-        id: nvm
-        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
-      - uses: actions/setup-node@v2
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - name: Install npm dependencies
+        run: yarn --immutable
+      - name: Run build script
+        run: yarn build:storybook
+      - name: Deploy to `storybook-static` directory of `gh-pages` branch
+        uses: peaceiris/actions-gh-pages@de7ea6f8efb354206b205ef54722213d99067935
         with:
-          node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
-      - run: yarn
-      - uses: MetaMask/action-publish-gh-pages@v2
-        with:
-          build-command: build-storybook
-          source-directory: storybook-static
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # This `PUBLISH_DOCS_TOKEN` needs to be manually set per-repository.
+          # Look in the repository settings under "Environments", and set this token in the `github-pages` environment.
+          personal_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
In the past we've used `GITHUB_TOKEN` for this, but we've slowly been creating more focused, restricted tokens for specific use cases. This seems similar to `PUBLISH_DOCS_TOKEN`, so I'm thinking maybe we want a `PUBLISH_STORYBOOK_TOKEN`. I've set up this token within the `github-pages` environment so we just need to change the name here:

``` suggestion
          # This `PUBLISH_STORYBOOK_TOKEN` needs to be manually set per-repository.
          # Look in the repository settings under "Environments", and set this token in the `github-pages` environment.
          personal_token: ${{ secrets.PUBLISH_STORYBOOK_TOKEN }}

Reply:
Thanks for the suggestion! Updated https://github.com/MetaMask/design-tokens/pull/601/commits/b9c10cda584ed4cde834477a7acfedb6421be720